### PR TITLE
Pull xarray's nbytes from nbytes attribute on arrays

### LINKED ISSUE #4842: nbytes does not return the true size for sparse variables
This wasn't entirely surprising to me, but `nbytes` currently doesn't return the right value for sparse data -- at least, I think nbytes should show the actual size in memory?

Since it uses `size` here:

https://github.com/pydata/xarray/blob/a0c71c1508f34345ad7eef244cdbbe224e031c1b/xarray/core/variable.py#L349

Rather than something like `data.nnz`, which of course only exists for sparse arrays...
I'm not sure if there's a sparse flag or something, or whether you'd have to do a typecheck?

**Minimal Complete Verifiable Example**:

```python
import pandas as pd
import numpy as np
import xarray as xr


df = pd.DataFrame()
df["x"] = np.repeat(np.random.rand(10_000), 10)
df["y"] = np.repeat(np.random.rand(10_000), 10)
df["time"] = np.tile(pd.date_range("2000-01-01", "2000-03-10", freq="W"), 10_000)
df["rate"] = 10.0
df = df.set_index(["time", "y", "x"])

sparse_ds = xr.Dataset.from_dataframe(df, sparse=True)
print(sparse_ds["rate"].nbytes)
```

```python
8000000000
```
**Anything else we need to know?**:

**Environment**:

<details><summary>Output of <tt>xr.show_versions()</tt></summary>

```
INSTALLED VERSIONS
------------------
commit: None
python: 3.7.9 (default, Aug 31 2020, 17:10:11) [MSC v.1916 64 bit (AMD64)]
python-bits: 64
OS: Windows
OS-release: 10
machine: AMD64
processor: Intel64 Family 6 Model 158 Stepping 9, GenuineIntel
byteorder: little
LC_ALL: None
LANG: en_US.UTF-8
LOCALE: None.None
libhdf5: 1.10.5
libnetcdf: 4.7.3

xarray: 0.16.1
pandas: 1.1.2
numpy: 1.19.1
scipy: 1.5.2
netCDF4: 1.5.3
pydap: None
h5netcdf: 0.8.0
h5py: 2.10.0
Nio: None
zarr: 2.4.0
cftime: 1.2.1
nc_time_axis: None
PseudoNetCDF: None
rasterio: 1.1.2
cfgrib: None
iris: None
bottleneck: 1.3.2
dask: 2.27.0
distributed: 2.30.1
matplotlib: 3.3.1
cartopy: None
seaborn: 0.11.0
numbagg: None
pint: None
setuptools: 49.6.0.post20201009
pip: 20.3.3
conda: None
pytest: 6.1.0
IPython: 7.19.0
sphinx: 3.2.1
```
</details>

### LINKED ISSUE #6565: `DataArray.nbytes` listed twice in API doc block
Here: https://github.com/pydata/xarray/blame/708204370ef168f35f250ca75e666246ae1420f6/doc/api.rst#L289-L295

### PR DESCRIPTION
**Pull request by maxrjones**
This implements the suggestion in https://github.com/pydata/xarray/issues/4842#issuecomment-769198581 to leave the nbytes calculation to the backend array.

Also removed the duplicate docs entry reported in #6565

- [x] Closes #6565
- [x] Closes #4842

### CODE REVIEW

**Comment by maxrjones** [comment]
Please let me know if I should add a docstring as suggested in https://github.com/pydata/xarray/issues/6565#issuecomment-1115545322. I didn't yet because most of the properties do not have docstrings.

**Comment by dcherian** [comment]
> I should add a docstring as suggested

This would be great!

Adding docstrings for the other properties would be a great (separate) PR!

File: xarray/core/variable.py [hunk 1]
```diff
@@ -333,7 +333,7 @@ def shape(self):
 
     @property
     def nbytes(self):
-        return self.size * self.dtype.itemsize
+        return self.data.nbytes
```
Comment (dcherian) [inline xarray/core/variable.py:336]: I think we'll want to account for `nbytes` not existing on `self.data` (it's not in the [array API](https://data-apis.org/array-api/latest/API_specification/array_object.html#attributes)) So something like

```
if hasattr(self.data, "nbytes"):
    ...
else:
    return self.size * self.dtype.itemsize
```

**Comment by dcherian** [comment]
Thanks @maxrjones can you push any WIP tests you might have for this?

**Comment by maxrjones** [comment]
> Thanks @maxrjones can you push any WIP tests you might have for this?

If it's alright I will wait for https://github.com/pydata/xarray/pull/6804 to be merged first because that will greatly simplify testing the changes. After that PR, the following could be added to `xarray/tests/test_array_api.py` to test both cases:

```
def test_properties(arrays) -> None:
    np_arr, xp_arr = arrays
    assert np_arr.nbytes == 48
    assert xp_arr.nbytes == 48
```

**Comment by dcherian** [comment]
Ah great idea!

**Comment by TomNicholas** [comment]
Great first contribution @maxrjones ! #6804 should auto-merge in a sec so you can then fetch and merge in `main` to write your tests :smile:

File: xarray/tests/test_array_api.py [hunk 2]
```diff
@@ -43,6 +43,12 @@ def test_indexing(arrays) -> None:
     assert_equal(actual, expected)
 
 
+def test_properties(arrays) -> None:
+    np_arr, xp_arr = arrays
+    assert np_arr.nbytes == 48
+    assert xp_arr.nbytes == 48
```
Comment (Illviljan) [inline xarray/tests/test_array_api.py:49]:
```suggestion
    assert np_arr.nbytes == np_arr.data.nbytes
    assert xp_arr.nbytes == np_arr.data.nbytes
```
I think at some point xp might not always be the array api and then the value could change.

File: xarray/core/variable.py [hunk 3]
```diff
@@ -335,7 +335,13 @@ def shape(self):
 
     @property
     def nbytes(self):
```
Comment (Illviljan) [inline xarray/core/variable.py:337]:
```suggestion
    def nbytes(self) -> int:
```
Can we add the typing here as well?

**Comment by dcherian** [comment]
Thanks @maxrjones I added a sparse variable test that fails on main.

**Review by dcherian — APPROVED** [review]
Thanks @maxrjones This is a great and impactful PR! Welcome to Xarray

**Pull request merged**